internal/metricdata/servicestats.go

```diff
@@ -0,0 +1,109 @@
+// SPDX-FileCopyrightText: 2022-present Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package metricdata
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/omec-project/metricfunc/logger"
+	"github.com/omec-project/metricfunc/pkg/metricinfo"
+)
+
+type nfServiceStats struct {
+	svcStatLock sync.RWMutex
+	svcStats    map[string]map[string]uint64 //Nf IP is key
+}
+
+func HandleServiceEvent(msgType *metricinfo.CoreMsgType, sourceNf metricinfo.NfType) {
+	switch sourceNf {
+	case metricinfo.NfTypeSmf:
+		handleSmfServiceEvent(msgType)
+	case metricinfo.NfTypeAmf:
+		handleAmfServiceEvent(msgType)
+	default:
+		logger.CacheLog.Errorf("unknown msg source [%v] ", sourceNf)
+	}
+}
+
+func handleSmfServiceEvent(msgType *metricinfo.CoreMsgType) {
+
+	metricData.SmfSvcStats.svcStatLock.Lock()
+	defer metricData.SmfSvcStats.svcStatLock.Unlock()
+
+	//Check if bucket already exist
+	if stats, ok := metricData.SmfSvcStats.svcStats[msgType.SourceNfIp]; ok {
+		stats[msgType.MsgType] = stats[msgType.MsgType] + 1
+		return
+	}
+
+	stat := make(map[string]uint64)
+	stat[msgType.MsgType] = 1
+	metricData.SmfSvcStats.svcStats[msgType.SourceNfIp] = stat
+
+	fmt.Printf("smf svc metric data content : %v ", metricData.SmfSvcStats.svcStats)
```

Comment:
Do you need this log ? If yes then we should not print for all instances..Print just new bucket created for ...

Reply:
This was just debug level, shall remove this log